docs/implementing-cards.md

```diff
@@ -1,388 +1,570 @@
+
+Copy-paste these values into the card impl file, and add a static class variable `<className>.implemented = true` to mark for the system that the card is implemented. The final result should look like below:
 
 ```javascript
-const DrawCard = require('../../drawcard.js');
-const AbilityDsl = require('../../abilitydsl.js');
+import AbilityHelper from '../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../core/card/NonLeaderUnitCard';
 
-class CloudTheMind extends DrawCard {
-    // Card definition
-}
+export default class GroguIrresistible extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '6536128825',
+            internalName: 'grogu#irresistible'
+        };
+    }
 
-CloudTheMind.id = 'cloud-the-mind';
+    // implementation here
+}
 
-module.exports = CloudTheMind;
+GroguIrresistible.implemented = true;
 ```
 
-#### 3. Override the `setupCardAbilities` method.
+## Implement card abilities (quickstart)
+
+The below is a quickstart guide on how to implement each ability type with some examples without going into too much detail on the components. See section **TODO: LINK HERE** for details on the building blocks of abilities.
 
-Persistent effects, actions, and triggered abilities should be defined in the `setupCardAbilities` method. Use the AbilityDsl import to get access to tools to help with  implementations. See below for more documentation.
+All card abilities (i.e., any card text with an effect) should be defined in the `setupCardAbilities` method:
 
 ```javascript
-class CloudTheMind extends DrawCard {
-    setupCardAbilities() {
-        // Declare persistent effects, reactions and interrupts here.
+class GroguIrresistible extends NonLeaderUnitCard {
+    public override setupCardAbilities() {
+        // Declare all ability types (action, triggered, constant, event, epic, replacement) here
+        this.addActionAbility({
+            title: 'Exhaust an enemy unit',
+            cost: AbilityHelper.costs.exhaustSelf(),
+            targetResolver: {
+                controller: RelativePlayer.Opponent,
+                immediateEffect: AbilityHelper.immediateEffects.exhaust()
+            }
+        });
     }
 }
 ```
 
-### Keywords
+There are several ability types in SWU, each with its own initialization method. As shown above in the example of an action ability, each method accepts a property object defining the ability's behavior. Use the `AbilityHelper` import to get access to tools to help with implementations. Additionally, see [Interfaces.ts](../server/game/Interfaces.ts) for a list of available parameters for each ability type.
 
-Most Keywords are automatically parsed from the card text. It isn't necessary to explicitly implement them unless they are provided by a conditional persistent effect.
+The ability types and methods are:
 
-### Static bonuses from attachments
+| Ability Type | Method | Definition |
+| --- | --- | --- |
```

Comment:
We should add an example column here with a couple varied examples

Reply:
Do you mean examples of abilities or examples of code implementations?

Reply:
@AMMayberry1 just like example cards for each one so people would know what to look